code/modules/reagents/chemistry/machinery/pandemic.dm

```diff
@@ -68,117 +79,223 @@
+				return
+			var/vaccine_type = GetResistancesByIndex(resistance_index)
+			var/vaccine_name = "Unknown"
+			if(!ispath(vaccine_type))
+				if(GLOB.archive_diseases[vaccine_type])
+					var/datum/disease/D = GLOB.archive_diseases[vaccine_type]
+					if(D)
+						vaccine_name = D.name
+			else if(vaccine_type)
+				var/datum/disease/D = new vaccine_type(0, null)
+				if(D)
+					vaccine_name = D.name
+
+			if(!vaccine_type)
+				atom_say("Unable to synthesize requested antibody.")
+				return
+
+			var/obj/item/reagent_containers/glass/bottle/B = create_culture(vaccine_name, "vaccine", 200)
+			B.reagents.add_reagent("vaccine", 15, list(vaccine_type))
+		if("eject_beaker")
+			eject_beaker()
+		if("destroy_eject_beaker")
+			beaker.reagents.clear_reagents()
+			eject_beaker()
+		if("print_release_forms")
+			var/strain_index = text2num(params["strain_index"])
+			if(isnull(strain_index))
+				atom_say("Unable to respond to command.")
+				return
+			var/type = GetVirusTypeByIndex(strain_index)
+			if(!type)
+				atom_say("Unable to find requested strain.")
+				return
+			var/datum/disease/advance/A = GLOB.archive_diseases[type]
+			if(!A)
+				atom_say("Unable to find requested strain.")
+				return
+			print_form(A, usr)
+		if("name_strain")
+			var/strain_index = text2num(params["strain_index"])
+			if(isnull(strain_index))
+				atom_say("Unable to respond to command.")
+				return
+			var/type = GetVirusTypeByIndex(strain_index)
+			if(!type)
+				atom_say("Unable to find requested strain.")
+				return
+			var/datum/disease/advance/A = GLOB.archive_diseases[type]
+			if(!A)
+				atom_say("Unable to find requested strain.")
+				return
+			if(A.name != "Unknown")
+				atom_say("Request rejected. Strain already has a name.")
+				return
+			var/new_name = tgui_input_text(usr, "Name the Strain", "New Name", max_length = MAX_NAME_LEN)
+			if(!new_name)
+				return
 			A.AssignName(new_name)
 			for(var/datum/disease/advance/AD in GLOB.active_diseases)
 				AD.Refresh()
-		updateUsrDialog()
-	else if(href_list["print_form"])
-		var/datum/disease/D = GetVirusByIndex(text2num(href_list["print_form"]))
-		D = GLOB.archive_diseases[D.GetDiseaseID()]//We know it's advanced no need to check
-		print_form(D, usr)
+		if("switch_strain")
+			var/strain_index = text2num(params["strain_index"])
+			if(isnull(strain_index) || strain_index < 1)
+				atom_say("Unable to respond to command.")
+				return
+			var/list/viruses = GetViruses()
+			if(strain_index > length(viruses))
+				atom_say("Unable to find requested strain.")
+				return
+			selected_strain_index = strain_index;
+		else
+			return FALSE
 
+/obj/machinery/computer/pandemic/ui_state(mob/user)
+	return GLOB.default_state
 
-	else
-		usr << browse(null, "window=pandemic")
-		updateUsrDialog()
-		return
+/obj/machinery/computer/pandemic/ui_interact(mob/user, datum/tgui/ui = null)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "PanDEMIC", name)
+		ui.open()
```

Comment:
```suggestion
		ui = new(user, src, "PanDEMIC", name)
		ui.set_autoupdate(FALSE)
		ui.open()
```
Too much information and all without static_data
Not sure if the interface needs to be automatically updated every second

Reply:
> That would require the user to open and close the UI every time they eject and reinsert a beaker

SStgui.update_uis(src) when insert's beaker
Any act in ui will update_ui and data, except static_data